valhalla/filesystem.h

```diff
@@ -308,4 +341,19 @@ inline bool remove(const path& p) {
   return ::remove(p.c_str()) == 0;
 }
 
+inline bool remove_all(const path& p) {
```

Comment:
heres where we implement `rm -r`. i think the comments suitably explain what is going on